deploy/kubernetes/setup-sumologic.yaml.tmpl

```diff
@@ -79,51 +83,56 @@ data:
           cluster = var.cluster_name
         }
     }
-  
-    resource "sumologic_http_source" "default_metrics_source" {
-        name         = local.default-metrics-source-name
+    
+    resource "sumologic_http_source" "apiserver_metrics_source" {
+        name         = local.apiserver_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
-    resource "sumologic_http_source" "apiserver_metrics_source" {
-        name         = local.apiserver-metrics-source-name
+    
+    resource "sumologic_http_source" "control_plane_metrics_source" {
+        name         = local.control_plane_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
-    resource "sumologic_http_source" "events_source" {
-        name         = local.events-source-name
-        category     = "${var.cluster_name}/${local.events-source-name}"
+    
+    resource "sumologic_http_source" "controller_metrics_source" {
+        name         = local.controller_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
-    resource "sumologic_http_source" "kube_controller_manager_metrics_source" {
-        name         = local.kube-controller-manager-metrics-source-name
+    
+    resource "sumologic_http_source" "default_metrics_source" {
+        name         = local.default_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
-    resource "sumologic_http_source" "kube_scheduler_metrics_source" {
-        name         = local.kube-scheduler-metrics-source-name
+    
+    resource "sumologic_http_source" "kubelet_metrics_source" {
+        name         = local.kubelet_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
-    resource "sumologic_http_source" "kube_state_metrics_source" {
-        name         = local.kube-state-metrics-source-name
+    
+    resource "sumologic_http_source" "node_metrics_source" {
+        name         = local.node_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
-    resource "sumologic_http_source" "kubelet_metrics_source" {
-        name         = local.kubelet-metrics-source-name
+    
+    resource "sumologic_http_source" "scheduler_metrics_source" {
+        name         = local.scheduler_metrics_source
         collector_id = "${sumologic_collector.collector.id}"
     }
-  
+    
+    resource "sumologic_http_source" "state_metrics_source" {
```

Comment:
Perhaps we do this in another PR, but I believe we need to expose the sourceCategory for all metrics sources.  Right now, we do not set it which results in all metrics getting the default sourceCageory of "Http Input". this makes data volume reporting tricky.  We should follow the same model as events for the metrics sources so that the data volume reporting in Sumo is managemble.

Reply:
Logs are a bit different, because logs sourceCategory is set via the HTTP header. This is not true for metrics and events which is why we need to be able to provide a sensible default.